### PR TITLE
Update to TypeScript 3.6.3

### LINKED ISSUE #2869: Upgrade to TypeScript 3.6
This is currently blocked by microsoft/TypeScript#33142 which a fix will be in 3.6.3, and while the fix will work, I still think it is not a full fix, but good enough for now.

Essentially they introduced code that assumes `process` is available in the runtime environment.

### PR DESCRIPTION
**Pull request by kitsonk**
Resolves #2869

### CODE REVIEW

**Comment by piscisaureus** [comment]
We also use typescript as submodule nowadays (in /deno_typescript/typescript). Do we still need to keep the copy in /third_party?
cc @ry

**Comment by ry** [comment]
I have tried to remove ts from package.json and third party, but it breaks on eslint. I couldn’t figure out why.

File: cli/tests/error_003_typescript.ts [hunk 1]
```diff
@@ -1,26 +1,20 @@
...
+let y = {
+  a: {
+    b: {
+      c() {
+        return { d: 1234 };
+      }
+    }
+  }
+};
+x = y;
```
Comment (ry) [inline cli/tests/error_003_typescript.ts]: 🙄

Comment (kitsonk) [reply cli/tests/error_003_typescript.ts]: Our previous test was fixed in 3.6.3, so I had to replace it and @DanielRosenwasser suggested this one (though it is likely to get less complex in 3.7, so...). Just wanted something obnoxious to test.

**Comment by kitsonk** [comment]
@ry `eslint` and related requires `typescript` to function properly, with it resolvable within `node_modules` 🤷‍♂ I will think about if there is any easy way to resolve that.

Yeah, I should have explained the diagnostics in the PR text better.  In 3.6, the diagnostic chain went from a single instance to an array, and the code that was there previously on the Rust side was pretty crap (I wrote it, so I can say that).  Having to adjust it, meant that I was able to revisit it and get it to be a bit cleaner.

**Pull request merged**